src/main/java/com/googleinterns/zoomtube/servlets/LectureServlet.java

```diff
@@ -60,15 +60,11 @@
   @VisibleForTesting static final String PARAM_LINK = "link-input";
   @VisibleForTesting static final String PARAM_ID = "id";
 
-  /* Pattern used to create a matcher for a video ID. */
-  private static Pattern videoUrlGeneratedPattern;
-
   private static DatastoreService datastore;
 
   @Override
   public void init() throws ServletException {
     datastore = DatastoreServiceFactory.getDatastoreService();
-    videoUrlGeneratedPattern = Pattern.compile(YOUTUBE_VIDEO_URL_PATTERN);
```

Comment:
We should set the pattern statically instead of setting it in `init`:
```
private static Pattern videoUrlGeneratedPattern = Pattern.compile(YOUTUBE_VIDEO_URL_PATTERN);
```